src/main/java/org/wso2/carbon/connector/connection/EmailConnection.java

```diff
@@ -0,0 +1,144 @@
+
+    public EmailConnection(ConnectionConfiguration connectionConfiguration) {
+
+        this.protocol = connectionConfiguration.getProtocol();
+        Properties sessionProperties = setSessionProperties(connectionConfiguration.getHost(),
+                connectionConfiguration.getPort());
+        sessionProperties.putAll(setTimeouts(connectionConfiguration.getReadTimeout(),
+                connectionConfiguration.getWriteTimeout(), connectionConfiguration.getConnectionTimeout()));
+
+        if (protocol.isSecure()) {
+            sessionProperties.putAll(setSecureProperties(connectionConfiguration));
+        }
+
+        this.session = Session.getInstance(sessionProperties,
+                new javax.mail.Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+
+                        return new PasswordAuthentication(connectionConfiguration.getUsername(),
+                                connectionConfiguration.getPassword());
+                    }
+                });
+    }
+
+    public Session getSession() {
+
+        return session;
+    }
+
+    /**
+     * Sets basic session properties required by the protocol
+     *
+     * @param host Host name of the server
+     * @param port Port to connect to
+     * @return Properties configured
+     */
+    private Properties setSessionProperties(String host, String port) {
+
+        Properties props = new Properties();
+        props.setProperty(protocol.getPortProperty(), port);
+        props.setProperty(protocol.getHostProperty(), host);
+        props.setProperty(protocol.getTransportProtocolProperty(), protocol.getName());
+        props.setProperty(protocol.getMailAuthProperty(), TRUE);
+        return props;
+    }
+
+    /**
+     * Sets secure properties
+     *
+     * @param connectionConfiguration configurations to be set
+     * @return Properties to be configured
+     */
+    private Properties setSecureProperties(ConnectionConfiguration connectionConfiguration) {
+
+        Properties props = new Properties();
+        props.setProperty(protocol.getStartTlsProperty(), TRUE);
+        if (connectionConfiguration.isRequireTLS()) {
+            props.setProperty(protocol.getStartTlsProperty(), TRUE);
+        } else {
+            props.setProperty(protocol.getSslEnableProperty(), TRUE);
+            props.setProperty(protocol.getSocketFactoryFallbackProperty(),
+                    EmailConstants.DEFAULT_SOCKETFACTORY_FALLBACK);
+            props.setProperty(protocol.getSocketFactoryPortProperty(),
+                    String.valueOf(connectionConfiguration.getPort()));
+        }
+
+        if (connectionConfiguration.getCipherSuites() != null) {
+            props.setProperty(protocol.getSslCipherSuitesProperty(),
+                    replaceWithWhitespace(connectionConfiguration.getCipherSuites()));
+        }
+
+        if (connectionConfiguration.getSslProtocols() != null) {
```

Comment:
Aren't these properties mandatory for creating a secured connection? If they are we need to fail the connector deployment by throwing error code "CONFIG_ERROR"?

Reply:
These are not mandatory parameters. It is only required to specify whether to use SSL or TLS which is configured using a template parameter requireTLS. The default will be false. Other parameters are optional.

Reply:
How do we know SSL or TLS is enabled? If that is the case, these become mandatory, aren't they?

Reply:
Based on the protocol Email Server supports, the user is expected to configure the 'requireTLS' template parameter in 'init'. Java Mail will use the default cipher suites, protocols etc. that it supports if this is not configured. These configurations are only required if it is required to configure a configuration such as a ciphersuite, which is not listed in the default list of cipher suites.

Reply:
+1. We will capture these details in the documentation.